ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRaoProvider.java

```diff
@@ -67,74 +69,97 @@ public SearchTreeRaoProvider() {
     public CompletableFuture<RaoResult> run(RaoInput raoInput, RaoParameters parameters) {
         RaoUtil.initData(raoInput, parameters);
 
+        // optimization is made on one given state only
         if (raoInput.getOptimizedState() != null) {
-            RaoData raoData = new RaoData(
-                    raoInput.getNetwork(),
-                    raoInput.getCrac(),
-                    raoInput.getOptimizedState(),
-                    raoInput.getPerimeter(),
-                    raoInput.getReferenceProgram(),
-                    raoInput.getGlskProvider(),
-                    raoInput.getBaseCracVariantId(),
-                    parameters.getLoopflowCountries());
-            TreeParameters treeParameters = TreeParameters.buildForPreventivePerimeter(parameters.getExtension(SearchTreeRaoParameters.class), true);
-            return CompletableFuture.completedFuture(new SearchTree().run(raoData, parameters, treeParameters).join());
+            return optimizeOneStateOnly(raoInput, parameters);
         }
 
+        // compute initial sensitivity on all CNECs
+        // this is necessary to have initial flows for MNEC and loopflow constraints on CNECs, in preventive and curative perimeters
+        SystematicSensitivityResult initialSensitivityResult;
+        try {
+            initialSensitivityResult = initialSensitivityAnalysisOnAllPerimeters(raoInput, parameters);
+        } catch (SensitivityAnalysisException e) {
```

Comment:
can you log the error here ?

Reply:
ok, i added a similar log in the LinearRao as well